app/shop.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+from typing import Any
+
+from app.point import Point
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            location: Point,
+            products: dict[str, float | int]
+    ) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+
+    @classmethod
+    def create_shop(
+            cls,
+            shop_info: dict[str, str | list[int] | dict[str, (int | float)]]
```

Comment:
The type hint for `shop_info` should be `dict[str, Union[str, list[int], dict[str, float]]]` instead of `dict[str, str | list[int] | dict[str, (int | float)]]` to match the expected types for `name`, `location`, and `products`.